### PR TITLE
Fix StrptimeLexer.flex to add a comment about how to generate the lexer 

### PR DESCRIPTION
**Pull request by muga**
This PR comes from #4635 and fixes StrptimeLexer.flex to add a comment for how to generate the lexer. 

Ideally we should not put those lexer java programs as @enebo and @headius discussed it on #4635 and will fix that on #4665.

### CODE REVIEW

File: core/src/main/java/org/jruby/lexer/StrptimeLexer.java [hunk 1]
```diff
@@ -10,7 +17,7 @@
 /**
  * This class is a scanner generated by 
  * <a href="http://www.jflex.de/">JFlex</a> 1.4
- * on 5/6/17 4:40 AM from the specification file
+ * on 6/16/17 3:03 PM from the specification file
  * <tt>core/src/main/java/org/jruby/lexer/StrptimeLexer.flex</tt>
  */
 public class StrptimeLexer {
```
Comment (muga) [inline core/src/main/java/org/jruby/lexer/StrptimeLexer.java]: I re-generated .java file. If we don't need the update, please let me know. I will remove it.

**Comment by muga** [comment]
@kares Thank you for reviewing and commenting. Yea, expactly, the latest JFlex is 1.6.1. As you know, the current `StrftimeLexer.flex` and `StrptimeLexer.flex` work with JFlex 1.4 only (Not work with 1.4.1,.. 1.6.1). It's better that the current .flex files are ideally refactored and fixed for JFlex 1.6.1 and then, use jflex-maven-plugin to automatically generate .java files. I think that it would be worked on #4665.

**Comment by muga** [comment]
I'm ignoring the CI failures because it seems that they're not related to this fix.

**Comment by headius** [comment]
Thank you for your continued improvements 👍

**Pull request merged**